account/accounts.go

```diff
@@ -477,3 +480,100 @@ func (m *Manager) MigrateKeyStoreDir(oldDir, newDir string, addresses []string)
 
 	return nil
 }
+
+func (m *Manager) ReEncryptKey(rawKey []byte, pass string, newPass string) (reEncryptedKey []byte, e error) {
+	cryptoJSON, e := keystore.RawKeyToCryptoJSON(rawKey)
+	if e != nil {
+		return reEncryptedKey, fmt.Errorf("convert to crypto json error: %v", e)
+	}
+
+	decryptedKey, e := keystore.DecryptKey(rawKey, pass)
+	if e != nil {
+		return reEncryptedKey, fmt.Errorf("decryption error: %v", e)
+	}
+
+	n := int(cryptoJSON.KDFParams["n"].(float64))
+	p := int(cryptoJSON.KDFParams["p"].(float64))
```

Comment:
index error if n or p ar not present for some reason. Could you point me to the generator KDF params?

Reply:
We use this [generator fn](https://github.com/status-im/status-go/blob/develop/account/generator/generator.go#L267) which calls [ImportSingleExtendedKey](https://github.com/status-im/go-ethereum/blob/v1.9.5-status.12/accounts/keystore/keystore.go#L469) and finally [importKey](https://github.com/status-im/go-ethereum/blob/v1.9.5-status.12/accounts/keystore/keystore.go#L522).

Can these `n` and `p` values be different depending on how the account was setup?

Reply:
Yes, there's plenty of ways to generate BIP44 keys and users might actually import them from unknown sources

Reply:
Added a check on existence of `n` and `p`:

```go
	if cryptoJSON.KDFParams["n"] == nil || cryptoJSON.KDFParams["p"] == nil {
		return reEncryptedKey, fmt.Errorf("Unable to determine `n` or `p`: %v", e)
	}
```
https://github.com/status-im/status-go/pull/2223/files#diff-0a5c50ab0de0200125520596dd297061369d42a646ede3641296789c621961afR495